### PR TITLE
feat: add hook use download

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Add hook useDownload to download files using blob objects

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Hooks/UseDownload/index.tsx [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default as useDownload } from './UseDownload'
```
Comment (AbhishekA1509) [inline src/Shared/Hooks/UseDownload/index.tsx:1]: I believe we should also export HandleDownloadProps

File: src/Shared/Hooks/UseDownload/UseDownload.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                    document.body.removeChild(a)
+                }, 0)
+
+                toast.success('Downloaded Successfully')
+            } else {
+                const error = (await response.json()).errors[0].userMessage
+                showError(error)
+            }
+        } catch (error) {
+            toast.error(error)
```
Comment (AbhishekA1509) [inline src/Shared/Hooks/UseDownload/UseDownload.tsx:62]: Can use showError

File: src/Shared/Hooks/UseDownload/UseDownload.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                a.click()
+
+                // Clean up the URL object after the download is complete
+                setTimeout(() => {
+                    URL.revokeObjectURL(blobUrl)
+                    document.body.removeChild(a)
+                }, 0)
+
+                toast.success('Downloaded Successfully')
+            } else {
```
Comment (AbhishekA1509) [inline src/Shared/Hooks/UseDownload/UseDownload.tsx:57]: Check if we can move this else into catch block itself?

File: src/Shared/Hooks/UseDownload/UseDownload.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            toast.info(
+                <ToastBody
+                    title="Preparing file for download"
+                    subtitle="File will be downloaded when it is available."
+                />,
+            )
+        }
+        try {
+            const response = await getDownloadResponse(downloadUrl)
+            if (response.status === 200) {
```
Comment (AbhishekA1509) [inline src/Shared/Hooks/UseDownload/UseDownload.tsx:33]: Can use from api status code enum

File: src/Shared/Hooks/UseDownload/UseDownload.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    downloadUrl: string
+    showFilePreparingToast?: boolean
+    fileName?: string
+}
+
+const useDownload = () => {
+    const [isDownloading, setIsDownloading] = useState<boolean>(false)
+
+    /*
+    downloadUrl: API url for downloading file
```
Comment (AbhishekA1509) [inline src/Shared/Hooks/UseDownload/UseDownload.tsx:17]: Please format

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/Shared/Hooks/UseDownload/UseDownload.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                setTimeout(() => {
+                    URL.revokeObjectURL(blobUrl)
+                    document.body.removeChild(a)
+                }, 0)
+
+                toast.success('Downloaded Successfully')
+            } else {
+                const jsonResponse = await response?.json()
+                const error = jsonResponse.errors[0].userMessage || jsonResponse.errors[0].internalMessage
+                showError(new Error(error))
```
Comment (AbhishekA1509) [inline src/Shared/Hooks/UseDownload/UseDownload.tsx:60]: Can we send jsonResponse itself in showError

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**